tests/FSharp.Compiler.ComponentTests/EmittedIL/StructGettersReadOnly.fs

```diff
@@ -0,0 +1,142 @@
+namespace FSharp.Compiler.ComponentTests.EmittedIL
+
+open Microsoft.FSharp.Core
+open Xunit
+open FSharp.Test.Compiler
+open FSharp.Test.ReflectionHelper
+
+module ``Struct getters readonly`` =
+
+    let structRecord =
+        FSharp
+            """
+            module Test
+
+            [<Struct>] type MyRecord = { MyField : int }
+            """
+
+    let nonStructRecord =
+        FSharp
+            """
+            module Test
+
+            type MyRecord = { MyField : int }
+            """
+
+    let structAnonRecord =
+        FSharp
+            """
+            module Test
+
+            let myRecord = struct {| MyField = 3 |}
+            """
+
+    let nonStructAnonRecord =
+        FSharp
+            """
+            module Test
+
+            let myRecord = {| MyField = 3 |}
+            """
+
+    let structNonRecord =
+        FSharp
+            """
+            module Test
+
+            [<Struct>]
+            type MyStruct =
+                val MyField: int
+            """
+
+    let structWithCustomGetter =
+        FSharp
+            """
+            module Test
+
+            [<Struct>]
+            type MyStruct =
+                val mutable x: int
+                member this.MyField with get () = this.x <- 4
+            """
+
+    [<Fact>]
+    let ``Struct record has readonly attribute on getter`` () =
+        structRecord
```

Comment:
As you wish, but for me it would be more readable to have the code right here in the test.

Reply:
I agree it is more readable inline. Don't want to duplicate it for the tests where it's reused but I can do it in the others.